java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/searcher/TestNGTestSearcher.java

```diff
@@ -43,7 +43,7 @@ public boolean isTestMethod(IMethod method) {
                 return false;
             }
             for (final String annotation : this.testMethodAnnotations) {
-                if (TestFrameworkUtils.hasAnnotation(method, annotation)) {
+                if (TestFrameworkUtils.hasAnnotation(method, annotation, false /*checkHierarchy*/)) {
```

Comment:
Actually you can simply overload `hasAnnotation`, to avoid changing these statements. 
```
boolean hasAnnotation(method, anno) {
  return hasAnnotation(method, anno, false);
}

boolean hasAnnotation(method, anno, boolean) {
 // ...
}
```